src/libsyntax/parse/parser.rs

```diff
@@ -2323,6 +2324,16 @@ impl<'a> Parser<'a> {
                         self.recover_stmt();
                     }
                 }
+                if self.check(&token::Comma) {
+                    let mut err = self.sess.span_diagnostic.mut_span_err(
+                        exp_span.to(self.prev_span),
+                        "cannot use a comma after struct expansion",
+                    );
+                    err.span_suggestion_short(self.span, "remove this comma", "".to_owned());
+                    err.note("the struct expansion should always be the last field");
```

Comment:
nitpick: It _must_ be the last field, not just _should_, right?

Reply:
Alright, thanks, I will add a test. Should I call it `issue-41834.rs` or `struct-expansion.rs`? (not sure what's the correct term for the ` ..default()`)